### PR TITLE
Flash: Add additional tests and increase code coverage

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Requires https://github.com/bosagora/agora/pull/1815 (ignore first commit)

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
@omerfirmak I hope there's not too many commits. :)

File: source/agora/test/Flash.d [hunk 1]
```diff
@@ -1072,3 +1072,143 @@ unittest
...
+    auto res = alice.openNewChannel(
+        utxo, Amount(1), Settle_1_Blocks, bob_pair.V);
+    assert(res.error == ErrorCode.RejectedFundingAmount, res.to!string);
+
+    // error on capacity too high
+    res = alice.openNewChannel(
+        utxo, Amount(1_000_000_000), Settle_1_Blocks, bob_pair.V);
+    assert(res.error == ErrorCode.RejectedFundingAmount, res.to!string);
+
+    // error on settle time not within limits
```
Comment (omerfirmak) [inline source/agora/test/Flash.d]: Shouldnt there be a corresponding "too low" test?

**Review by omerfirmak — APPROVED** [review]
Glad I had my diving gear ready.

File: source/agora/test/Flash.d (hunk 1 above)
Comment (AndrejMitrovic) [reply source/agora/test/Flash.d]: I'll add it.

**Pull request merged**